debezium-core/src/main/java/io/debezium/relational/history/AbstractDatabaseHistory.java

```diff
@@ -57,9 +63,13 @@ public final void recover(Map<String, ?> source, Map<String, ?> position, Tables
                 String ddl = recovered.ddl();
                 if (ddl != null) {
                     ddlParser.setCurrentSchema(recovered.databaseName()); // may be null
+                    if (ddlFilter.test(ddl)) {
```

Comment:
I'm surprised it's applied ion `recover()`. Shouldn't it be done in `record()`?

Reply:
This is intentional - just in case a bad statement gets into the history topic. Otherwise the filtering is done already at `BinlogReader` level.